src/components/TradeWidget/index.tsx

```diff
@@ -106,18 +113,41 @@ const TradeWidget: React.FC = () => {
 
   let sameToken = sellToken === receiveToken
 
+  function onSubmit(data: FieldValues): void {
+    if (isConnected) {
+      placeOrder({
+        buyAmount: parseAmount(data[receiveInputId], receiveToken.decimals),
+        buyToken: receiveToken,
+        sellAmount: parseAmount(data[sellInputId], sellToken.decimals),
+        sellToken,
+      })
+    } else {
+      history.push('/connect-wallet')
```

Comment:
Not for this PR, but should we encode the return page in the URL of `connect-wallet`, so on success it goes back to the trade page and it prefills all the inputs again.

Alternatively, this would open a modal for connecting the wallet, so you don't even need to navigate.

Whatever makes it easier.

Reply:
NVM: I see you dealt with this in https://github.com/gnosis/dex-react/pull/184/files
This is great!